src/app/Header/Header.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.logout();
```

Comment:
why are you calling it here??????
everytime you load component this ngOninit gets called
hence your token will be deleted.
remove this